webapp/src/pages/offer/components/OfferComponent.js

```diff
@@ -0,0 +1,214 @@
+
+      <form class="pt-5 ">
+        <p class="font-weight-bold">You have been accepted as a  {userProfile!=null ? userProfile.user_category: "<Category>"} </p>
+        <p class="font-weight-bold">Your status</p>
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Travel Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.travel_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row">
+          <div class="col-8 font-weight-bold">
+            Accommodation Award:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.accomodation_award
+          : "Not available"}
+          </div>
+        </div>
+
+        <div class="row pb-5 ">
+          <div class="col-8 font-weight-bold">
+            Payment Required:
+          </div>
+          <div class="col-2">
+          {offerList!=null
+          ? offerList.payment_required
+          : "Not available"}
+          </div>
+        </div>
+        <p class="font-weight-bold">Please accept or reject this offer by {offerList!=null ? offerList.expiry_date : "unable to load expiry date"} </p>
+        <div class="row">
+         <div class="col">
+          <button type="button" class="btn btn-danger" id="reject"onClick={
+            ()=>{
+              this.setState({
+                rejected:true,
+                showReasonBox:true
+              },
+              this.buttonSubmit());
+            }
+          } >
+              Reject
+          </button>
+          </div>
+          <div class="col">
+          <button type="button" class="btn btn-success" id="accept" onClick={()=>{
+              this.setState({
+                accepted:true
+              });
+              this.buttonSubmit();
+          }}>
+              Accept
+          </button>
+          </div>
+        </div> 
+
+        <div class="form-group">
+         { this.state.showReasonBox ? 
+          <div class="form-group mr-5  ml-5 pt-5" >
+          <textarea class="form-control pr-5 pl-10" id="textArea" onChange={()=>{this.handleChange(this.rejected_reason)}}  placeholder="Enter rejection message"></textarea>
+          <button type="button" class="btn" id="apply" onClick={ ()=>{
+              this.setState({
+                showReasonBox:false
+              },
+              this.buttonSubmit());
+            }}>apply</button>
+        </div>
+        :""
+        }
+        </div>
+      
+      </form>
+    </div>
+      )}
+
+    componentDidMount(){
+      userService.get().then(result=> {
+        this.setState({
+            user:{
+              user_id:result.response.data.app_user_id
+            }
+        });
+      });
+
+    }
+
+    componentWillMount() {
+      const {user_id} = this.state.user;
+          this.getOffer();
+          this.setState({
+            user: JSON.parse(localStorage.getItem("user")),
```

Comment:
here you load the user from storage. 
You should actually just use this. the user.id is here already?